docs/docs/tutorials/your_own_llm/1_creating_your_dataset.md

```diff
@@ -0,0 +1,64 @@
+
+5. Domain-Specific Apps or Games: Developing small applications or games that capture specific language patterns or behaviors can be an interactive and engaging way to collect data. These apps can prompt users to generate text, provide feedback, or engage in dialogue, resulting in a dataset that aligns with your application's requirements.
+
+
+**Example: Alpaca Cleaned Format**
+
+The [Alpaca Cleaned](https://github.com/gururise/AlpacaDataCleaned) format is a dataset used in the Jac NLP's LLM module. It contains instructions from various domains, cleaned and organized for training LLMs. The format typically includes text examples with corresponding labels or metadata if applicable. For instance:
+
+```json
+[
+    {
+        "instruction": "Give three tips for staying healthy.",
+        "input": "",
+        "output": "1.Eat a balanced diet and make sure to include plenty of fruits and vegetables. \n2. Exercise regularly to keep your body active and strong. \n3. Get enough sleep and maintain a consistent sleep schedule."
+    },
+    {
+        "instruction": "Convert the following sentence into the present continuous tense",
+        "input": "He reads books",
+        "output": "He is reading books."
+    }
+]
+```
+
+The Alpaca Cleaned format provides a structured representation of instructions with associated domains and labels. This format facilitates the training process and ensures that the LLM can learn and generate text within specific domains or tasks.
+
+When creating your dataset, consider the specific requirements of your application and the type of language patterns you want your LLM to capture. Design your data collection or annotation process accordingly to ensure that the dataset is diverse, representative, and aligns with the task your LLM will perform.
+
+Remember to properly clean and preprocess the dataset by removing irrelevant or noisy text, standardizing formats, and ensuring data consistency. This prepares the dataset for effective training and improves the overall performance of your LLM.
+
+## Creating a Dataset for your LLM
+
+The Jaseci platform provides a convenient way to create and manage datasets for your LLM. Run the following command in JSCTL to run the GUI. Make sure to have jaseci, jac_nlp[llm], streamlit are installed. if not, run `pip install jaseci jac_nlp[llm] streamlit` to install them.
+
+```bash
+jaseci> actions load module jac_nlp.llm
+jaseci> actions call llm.run_dataset_builder
+
+  You can now view your Streamlit app in your browser.
+
+  Network URL: http://172.31.46.187:8501
+  External URL: http://34.211.22.126:8501
+
+# You can use the Network or External URL to access the GUI from your browser. Press Ctrl+C to stop the GUI. and exit the JSCTL shell when you are done.
+```
+
```

Comment:
provide an example of creating the dataset. At this point, it's not very clear.

Reply:
The application is self explantory. Were you able to run the app?

Reply:
Can run the app. Better if we have an ss with example input for a clear picture for users.